### PR TITLE
Lakhveer/callbacks

### PR DESCRIPTION
**Pull request by lakhveerk**
In appWindow, the postMessage functions use promise and wait for a response before triggering their callback, but these promises never get resolved. Reverting this back to callbacks like we had in v1.

### CODE REVIEW

File: change/@microsoft-teams-js-2e19994e-bf6c-4039-96ba-e01f9c9387e0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "In `appWindow.ts` file, `ChildAppWindow` and `ParentAppWindow` are convereted back to synchronous calls because the promise was never being resolved.",
```
Comment (TrevorJoelHarris) [inline change/@microsoft-teams-js-2e19994e-bf6c-4039-96ba-e01f9c9387e0.json:3]: extra 'e' in converted

File: packages/teams-js/src/public/appWindow.ts [hunk 2]
```diff
@@ -36,39 +31,20 @@ export interface IAppWindow {
...
-  }
-  public postMessageHelper(message: any): Promise<void> {
-    return new Promise<void>(resolve => {
-      resolve(send('messageForChild', message));
-    });
+    sendMessageToParent('messageForChild', [message], onComplete ? onComplete : getGenericOnCompleteHandler());
   }
-
+  /**
+   * Add a listener that will be called when an event is received from this AppWindow.
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/appWindow.ts:43]: Would this more correctly be "Add a listener that will be called when an event is received from the ChildAppWindow"?

File: packages/teams-js/src/public/appWindow.ts [hunk 3]
```diff
@@ -85,39 +61,21 @@ export class ParentAppWindow implements IAppWindow {
...
-  }
-  public postMessageHelper(message: any): Promise<void> {
-    return new Promise<void>(resolve => {
-      resolve(send('messageForParent', message));
-    });
+    sendMessageToParent('messageForParent', [message], onComplete ? onComplete : getGenericOnCompleteHandler());
   }
 
+  /**
+   * Add a listener that will be called when an event is received from this AppWindow.
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/appWindow.ts:74]: Would it be more correct to say "Add a listener that will be called when an event is received from the ParentAppWindow"?

File: packages/teams-js/src/public/appWindow.ts [hunk 4]
```diff
@@ -36,39 +31,20 @@ export interface IAppWindow {
...
-  /**
-   * @deprecated
-   * As of 2.0.0-beta.1, please use {@link ChildAppWindow.postMessage ChildAppWindow.postMessage(message: any): Promise\<void\>} instead.
-   * @param message - The message to send
-   * @param onComplete - The deprecated way of invoking a callback to know if the postMessage has been success/failed.
+   * @param onComplete - The callback to know if the postMessage has been success/failed.
    */
-  public postMessage(message: any, onComplete: (status: boolean, reason?: string) => void): void;
-  public postMessage(message: any, onComplete?: (status: boolean, reason?: string) => void): Promise<void> {
+  public postMessage(message: any, onComplete?: (status: boolean, reason?: string) => void): void {
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/appWindow.ts:38]: should onComplete be a required parameter? It looks like in the old version of this class it was required, so I was curious about whether what you are doing on line 40 (getGenericOnCompleteHandler) was actually changing how this worked?

Comment (lakhveerk) [reply packages/teams-js/src/public/appWindow.ts:38]: I cannot seem to find a release where it was required. Do you remember what release was it?

Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/appWindow.ts:38]: I was going off of the old signature (line 49 before your change) where onComplete was a required parameter

Comment (lakhveerk) [reply packages/teams-js/src/public/appWindow.ts:38]: This was optional in v1, like we talked, we are going to keep it optional going forward.

**Review by TrevorJoelHarris — APPROVED** [review]

**Pull request merged**